workspaces/adventure-pack/goodies/java/src/traverse_post_order/AP.java

```diff
@@ -0,0 +1,63 @@
+package traverse_post_order;
+
+import iterable_stream.IterableStream;
+import java.util.NoSuchElementException;
+import java.util.Stack;
+import pair.Pair;
+import simple_iterator.SimpleIterator;
+
+public final class AP {
+
+  /**
+   * Get {@link IterableStream iterable stream} of all the {@link TreeNode nodes} in the binary tree
+   * in post-order.
+   * @param root Binary tree root {@link TreeNode node}.
+   * @return The {@link IterableStream iterable stream} in post-order.
+   */
+  public static IterableStream<TreeNode> traversePostOrder(
+    final TreeNode root
+  ) {
+    final Stack<Pair<TreeNode, Boolean>> stack = new Stack<
```

Comment:
I'd be curious to get your thoughts on this as well @utsavatwork, but I've recently been trying to adopt the [`var` keyword in Java](https://docs.oracle.com/en/java/javase/17/language/local-variable-type-inference.html).

Similar to how TypeScript and other languages can infer a lot of types, Java can now too, based on the expression. So I think the stack can become `final var stack` since the expression on the right-hand side is already enough information for the compiler. Likewise when we pop from the stack, the variable type is also already known (thanks to the stack type), so I'd recommend using `var` there.

Reply:
hey, I had noticed this in the code for java inorder-traversal goodie. I kinda figured it out there but was hesitating to use as I had never used it before. But, as you explained, it completely makes sense to use "var" for local variables. I've made changes accordingly.